### ISSUE TITLE
collection: Regression in GitHub workflows after merging PR 610

### ISSUE THREAD

**Issue by berndfinger** [sap-linuxlab/community.sap_install#621]
The error message is:
```
[Invalid workflow file: .github/workflows/ansible-lint-sap_maintain_etc_hosts.yml#L1](https://github.com/sap-linuxlab/community.sap_install/actions/runs/7651827742/workflow)
No steps defined in `steps` and no workflow called in `uses` for the following jobs: ansible-lint
```

**Comment by berndfinger** [comment]
After a first look, it seems that the steps: section is required. @Wabri - Can you please have a look?
Here's a link to one of the failing action: https://github.com/sap-linuxlab/community.sap_install/actions/runs/7652044239
And a direct link to the changed files of PR #610: https://github.com/sap-linuxlab/community.sap_install/pull/610/files

**Comment by berndfinger** [comment]
The `steps:` section is however in [.github/workflows/ansible-lint-sap_reusable_workflow.yml](https://github.com/sap-linuxlab/community.sap_install/pull/610/files#diff-db3bd2ab398f57a7c676e6671a3d22abe9f086d80724d15a569410538f2466be).

**Comment by ja9fuchs** [comment]
Currently the reusable workflow file does not yet exist in the `main` branch.

That is probably the reason why the following
> uses: sap-linuxlab/community.sap_install/.github/workflows/ansible-lint-sap_reusable_workflow.yml**@main**

is not valid and causes an error. 

I suggest merging this new reusable workflow file into `main` first in a separate PR.

**Comment by Wabri** [comment]
Yes, I pointed it out this problem in this comment https://github.com/sap-linuxlab/community.sap_install/pull/610#issuecomment-1898190958

Maybe now we should try to run one of the action that went into an error

**Comment by berndfinger** [comment]
> Yes, I pointed it out this problem in this comment [#610 (comment)](https://github.com/sap-linuxlab/community.sap_install/pull/610#issuecomment-1898190958)

Thanks! I had read it but didn't pay enough attention to it.

> Maybe now we should try to run one of the action that went into that error and check if is ok

The file is now part of the main branch. I create a test role with a GitHub reusable workflow, see https://github.com/berndfinger/testrole01/tree/dev/.github/workflows. None of my attempts to switch from a normal to a reusable workflow was successful so far.

@Wabri Can you please share an example of a configuration which has worked for you?

**Comment by berndfinger** [comment]
Finally now working in my example, see https://github.com/berndfinger/testrole01/actions/runs/7667462910 .

**Comment by Wabri** [comment]
Awesome 🚀

**Comment by berndfinger** [comment]
@Wabri Unfortunately, it only helped in my test case. Now working on:
- trying to get the actions to work in my fork. Maybe I can use a variable for the GitHub organization.
- getting a little code change into dev, to verify that the actions are actually working.
- once that is done, implement the manual trigger for all workflows so that we can better test them.

From looking at https://github.com/Wabri/community.sap_install/actions, it seems the new reusable workflow hasn't worked for you. Are you still confident that it will work for this upstream repo?

**Comment by berndfinger** [comment]
@Wabri Maybe I solved it, see PR #627 . I set you as one of the reviewers.

**Comment by Wabri** [comment]
It works for me: https://github.com/Wabri/community.sap_install/actions/runs/7671846479

**Comment by Wabri** [comment]
Saw the problem, I put too many layers in the jobs section!

**Comment by berndfinger** [comment]
Hopefully solved by https://github.com/sap-linuxlab/community.sap_install/pull/627. Let's wait and see... :-)

**Comment by berndfinger** [comment]
I think we can close this issue now. No more problems showed up.
